src/app/debug/Debug.tsx

```diff
@@ -0,0 +1,321 @@
+import { accountsSelector, currentAccountSelector } from 'app/accounts/selectors';
+import { BungieError, HttpStatusError } from 'app/bungie-api/http-client';
+import { Token, getToken } from 'app/bungie-api/oauth-tokens';
+import { clarityCharacterStatsSelector, clarityDescriptionsSelector } from 'app/clarity/selectors';
+import { DimAuthToken, getToken as getDimApiToken } from 'app/dim-api/dim-api-helper';
+import {
+  apiPermissionGrantedSelector,
+  currentProfileSelector,
+  dimSyncErrorSelector,
+  settingSelector,
+  updateQueueLengthSelector,
+} from 'app/dim-api/selectors';
+import { t } from 'app/i18next-t';
+import { profileErrorSelector, profileResponseSelector } from 'app/inventory/selectors';
+import { useLoadStores } from 'app/inventory/store/hooks';
+import { TroubleshootingSettings } from 'app/settings/Troubleshooting';
+import { systemInfo } from 'app/shell/About';
+import LocalStorageInfo from 'app/storage/LocalStorageInfo';
+import { set } from 'app/storage/idb-keyval';
+import { DimError } from 'app/utils/dim-error';
+import { usePageTitle } from 'app/utils/hooks';
+import { wishListsLastFetchedSelector } from 'app/wishlists/selectors';
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import styles from './Debug.m.scss';
+
+/**
+ * A user-facing debug page that displays information about the DIM environment,
+ * browser, and various states and capabilities to help us figure things out.
+ *
+ * The text on this page is for DIM developers and does not need to be translated.
+ */
+export default function Debug() {
+  usePageTitle('Debug');
+  const bungieToken = getToken();
+  const dimApiToken = getDimApiToken();
+
+  const accounts = useSelector(accountsSelector);
+  const currentAccount = useSelector(currentAccountSelector);
+  const profileResponse = useSelector(profileResponseSelector);
+  const profileError = useSelector(profileErrorSelector);
+  const dimSyncProfile = useSelector(currentProfileSelector);
+  const dimSyncError = useSelector(dimSyncErrorSelector);
+  const dimSyncUpdateQueueSize = useSelector(updateQueueLengthSelector);
+  const apiPermissionGranted = useSelector(apiPermissionGrantedSelector);
+  const wishListsLastFetched = useSelector(wishListsLastFetchedSelector);
+  const wishlistSource = useSelector(settingSelector('wishListSource'));
+  const clarityDescriptions = useSelector(clarityDescriptionsSelector);
+  const clarityCharacterStats = useSelector(clarityCharacterStatsSelector);
+
+  useLoadStores(currentAccount);
+
+  const [localStorageError, setLocalStorageError] = useState<Error>();
+  const [idbError, setIdbError] = useState<Error>();
+  useEffect(() => {
+    (async () => {
+      try {
+        await set('idb-test', true);
+      } catch (e) {
+        setIdbError(e);
+      }
+    })();
+  }, []);
+
+  useEffect(() => {
+    try {
+      localStorage.setItem('test', 'true');
+    } catch (e) {
+      setLocalStorageError(e);
+    }
+  }, []);
```

Comment:
Not sure if this can ever be reached with broken IDB/localstorage because Index.tsx blocks on both working and just renders `StorageBroken` if they're not. I guess if the browser's storage service works at startup and crashes later while DIM is running?

Reply:
Yeah I don't think this can be reached right now, but I felt like if I was going to have a storage debug panel I ought to have these.